src/ZendCodingStandard/Sniffs/Commenting/FileLevelDocBlockSniff.php

```diff
@@ -0,0 +1,295 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-coding-standard for the canonical source repository
+ * @copyright https://github.com/zendframework/zend-coding-standard/blob/master/COPYING.md Copyright
+ * @license   https://github.com/zendframework/zend-coding-standard/blob/master/LICENSE.md New BSD License
+ */
+
+use Zend\CodingStandard\Utils\LicenseUtils;
+
+class ZendCodingStandard_Sniffs_Commenting_FileLevelDocBlockSniff implements \PHP_CodeSniffer_Sniff
+{
+    private $repo;
+
+    private $licenseFirstYear;
+
+    private $licenseLastYear;
+
+    private $copyrightChanged;
+
+    private $fix;
+
+    public function __construct()
+    {
+        $this->licenseFirstYear = gmdate('Y');
+        $this->licenseLastYear = gmdate('Y');
+
+        $year = LicenseUtils::getCopyrightFirstYear(LicenseUtils::getCopyrightFile());
+        if ($year !== null && $year < $this->licenseFirstYear) {
+            $this->licenseFirstYear = $year;
+        }
+
+        $year = LicenseUtils::getCopyrightFirstYear(LicenseUtils::getLicenseFile());
+        if ($year !== null && $year < $this->licenseFirstYear) {
+            $this->licenseFirstYear = $year;
+        }
+
+        $content = file_get_contents('composer.json');
+        $content = json_decode($content, true);
+
+        $this->repo = $content['name'];
+    }
+
+    /**
+     * Registers the tokens that this sniff wants to listen for.
+     *
+     * @return int[]
+     * @see    Tokens.php
+     */
+    public function register()
+    {
+        return [T_OPEN_TAG];
+    }
+
+    /**
+     * Called when one of the token types that this sniff is listening for is
+     * found.
+     *
+     * @param PHP_CodeSniffer_File $phpcsFile The PHP_CodeSniffer file where the
+     *                                        token was found.
+     * @param int                  $stackPtr  The position in the PHP_CodeSniffer
+     *                                        file's token stack where the token
+     *                                        was found.
+     *
+     * @return int Optionally returns a stack pointer. The sniff will not be
+     *                  called again on the current file until the returned stack
+     *                  pointer is reached. Return (count($tokens) + 1) to skip
+     *                  the rest of the file.
+     */
```

Comment:
Nitpick: I've been _not_ aligning types/names in annotations, and indenting any line continuations only by four characters. That allows more text per continuation line, and is easier to format overall (changes in types and/or variable name lengths then do not require additional formatting changes).